### PR TITLE
Improve the use_env_repo checkbox.

### PR DESCRIPTION
**Pull request by bcolferzd**
* Provide a link to the repo and the project.env.erb file for the project 

/cc @zendesk/samson

### Tasks
 - [ ] :+1: from team

### References
 - Jira link: https://zendesk.atlassian.net/browse/BRE-1208

### Risks
- Level: Low/Med/High

### CODE REVIEW

File: plugins/env/app/views/samson_env/_fields.html.erb [hunk 1]
```diff
@@ -5,9 +5,11 @@
   </legend>
 
   <% if ENV['DEPLOYMENT_ENV_REPO'] %>
-    <%= form.input :use_env_repo, label: "Use GitHub repository #{ENV['DEPLOYMENT_ENV_REPO']} to manage deployment environment variables",
-                   as: :check_box
-    %>
+    <% project = form.object %>
```
Comment (grosser) [inline plugins/env/app/views/samson_env/_fields.html.erb]: can use `@project`

File: plugins/env/app/views/samson_env/_fields.html.erb [hunk 2]
```diff
@@ -5,9 +5,11 @@
...
+    <%= check_box_tag(:use_env_repo) %>
+    <% link_target =  "#{ENV['DEPLOYMENT_ENV_REPO']}/#{project.permalink}.env.erb" %>
+    <% label_text = "Use the GitHub repo for deployment env vars for this project here #{link_target}" %>
+    <%= label_tag(:use_env_repo, link_to(label_text,link_target)) %>
```
Comment (grosser) [inline plugins/env/app/views/samson_env/_fields.html.erb]: `label:` with html_safe should also work

File: plugins/env/test/samson_env/samson_plugin_test.rb [hunk 3]
```diff
@@ -199,6 +201,17 @@ def render_view
...
         result.wont_include checkbox
       end
+
+      it 'renders project erb link when DEPLOYMENT_ENV_REPO is present' do
+        with_env DEPLOYMENT_ENV_REPO: dep_env_repo do
+          result = render_view
+          result.must_include repo_link
+        end
+      end
+      it 'does not renders project erb link when DEPLOYMENT_ENV_REPO is nil' do
```
Comment (grosser) [inline plugins/env/test/samson_env/samson_plugin_test.rb]: newline between `it`s

File: plugins/env/test/samson_env/samson_plugin_test.rb [hunk 4]
```diff
@@ -199,6 +201,17 @@ def render_view
...
+        result = render_view
+        result.wont_include repo_link
```
Comment (grosser) [inline plugins/env/test/samson_env/samson_plugin_test.rb]: can single-line that

**Comment by grosser** [comment]
screenshot for UI changes

**Comment by grosser** [comment]
also linking to a file that does not exists confusing

File: plugins/env/test/samson_env/samson_plugin_test.rb [hunk 5]
```diff
@@ -174,7 +174,9 @@ def protect_against_forgery?
     end
 
     describe 'project_form callback' do
-      let(:checkbox) { 'id="project_use_env_repo" />' }
+      let(:checkbox) { 'id="use_env_repo"' }
+      let(:dep_env_repo) { 'zendesk/test.git' }
```
Comment (grosser) [inline plugins/env/test/samson_env/samson_plugin_test.rb]: should not link to .git

**Comment by grosser** [comment]
modified it:

before:
screen shot 2018-12-25 at 10 47 55 am

after:
screen shot 2018-12-25 at 10 57 16 am

**Comment by grosser** [comment]
updated screenshot

File: plugins/env/test/samson_env/samson_plugin_test.rb (hunk 4 above)
Comment (bcolferzd) [reply plugins/env/test/samson_env/samson_plugin_test.rb]: Accepted.

File: plugins/env/test/samson_env/samson_plugin_test.rb (hunk 3 above)
Comment (bcolferzd) [reply plugins/env/test/samson_env/samson_plugin_test.rb]: Done ... don't know why rubocop didn't pick it up?

File: plugins/env/app/views/samson_env/_fields.html.erb (hunk 2 above)
Comment (bcolferzd) [reply plugins/env/app/views/samson_env/_fields.html.erb]: Done

File: plugins/env/app/views/samson_env/_fields.html.erb (hunk 1 above)
Comment (bcolferzd) [reply plugins/env/app/views/samson_env/_fields.html.erb]: Done but should I also have changed this for the other code blocks?

File: plugins/env/test/samson_env/samson_plugin_test.rb (hunk 5 above)
Comment (bcolferzd) [reply plugins/env/test/samson_env/samson_plugin_test.rb]: Done ... and maybe a check for this in the future when setting the Samson env for the repo or strip off the `.git` if it exists.

**Comment by grosser** [comment]
did you force-push away my commits on purpose ?

**Comment by grosser** [comment]
-> grosser/alternative

**Pull request merged**